### ISSUE TITLE
brace-style: don't warn about one-line form without braces

### ISSUE THREAD

**Issue by mourner** [eslint/eslint#1739]
`brace-style` rule should warn about incorrect brace style only when braces are actually present, since brace presence is controlled by another rule (`curly`).

I want to enforce a brace style while also supporting no-braces one-line form. Sample code:

```js
if (tag === 1) glyph.id = pbf.readVarint();
else if (tag === 2) glyph.bitmap = pbf.readBytes();
else if (tag === 3) glyph.width = pbf.readVarint();
else if (tag === 4) glyph.height = pbf.readVarint();
```

`"brace-style": [2, "1tbs", {"allowSingleLine": true}]` will say:

```
Closing curly brace does not appear on the same line as the subsequent block
```

**Comment by nzakas** [comment]
Yeah, that seems like a bug. You can't check braces style when there are no braces.

**Comment by nzakas** [comment]
Working on this.

**Comment by mourner** [comment]
@nzakas it's fixed for most cases but still reports for this one:

```js
    if (tag === 1) fontstack.name = pbf.readString();
    else if (tag === 2) fontstack.range = pbf.readString();
    else if (tag === 3) {
        var glyph = pbf.readMessage(readGlyph, {});
        fontstack.glyphs[glyph.id] = glyph;
    }
```

Is this intentional or also a bug?

**Comment by nzakas** [comment]
Hard to tell since I don't know what line is 19 in your example.

**Comment by mourner** [comment]
@nzakas the first line with the curly brace. So groups without braces pass, groups with braces on each if/else block pass, but groups with mixed style don't. Looks like a bug to me.

**Comment by mourner** [comment]
Awesome, thanks a lot!
